gateways/http/transfer/transfer.go

```diff
@@ -32,25 +40,90 @@ func Transfer(serv *mux.Router, usecase transfer.UseCase) *Handler {
 // @Accept  json
 // @Produce  json
 // @Header 201 {string} Token "request-id"
-// @Router /transfer [post]
+// @Router /transfers [post]
 func (h Handler) CreateTransfer(w http.ResponseWriter, r *http.Request) {
-	var tr transfer.TransferInput
+	var tr TransferInput
 	accountID, ok := middleware.GetAccountID(r.Context())
 	if !ok || accountID == "" {
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(r.Response.StatusCode)
 		return
 	}
+
 	decoder := json.NewDecoder(r.Body)
 	err := decoder.Decode(&tr)
+
+	if err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(r.Response.StatusCode)
+		return
+	}
+	accountOrigin, errOrigin := h.account.GetAccountByCPF(accountID)
```

Comment:
why is the handler orchestrating what needs to happen in a transfer, instead of the usecase?

Reply:
In this case, because the GetAccountByCPF method is a method pertaining to the account and not the transfer. As the usecases should not communicate, only in the outermost layer, I put these calls in handlers

Reply:
does this make sense? 🤨

Reply:
@Francisleide yeah, I see your point. let's discuss this later and then see how you could implement it differently, okay?

Reply:
Okay! TY!